RoosterPlanner.UI/src/app/pages/home/home.component.ts

```diff
@@ -3,31 +3,91 @@ import {MatDialog} from "@angular/material/dialog";
+import {ToastrService} from "ngx-toastr";
+import {Router} from "@angular/router";
 
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.scss']
 })
 export class HomeComponent implements OnInit {
-  projects: Project[] = [];
+
   loaded: boolean = false;
+  participations: Participation[] = [];
+  currentUser: User;
+  selectedProjects: Project[];
 
 
-  constructor(public dialog: MatDialog, private projectService: ProjectService) {
+  constructor(public dialog: MatDialog,
+              private projectService: ProjectService,
+              private userService: UserService,
+              private participationService: ParticipationService,
+              private toastr: ToastrService,
+              private route: Router) {
   }
 
   async ngOnInit(): Promise<void> {
-    await this.projectService.getProject().then(response => {
-      this.projects = response;
-      this.loaded = true;
+    await this.userService.getCurrentUser().then(async user => {
+      if (user !== false) {
+        this.currentUser = user
+        this.getParticipations().then(x => this.loaded = true)
+      }
     });
   }
 
-  addProject() {
-    let dialogRef = this.dialog.open(AddProjectComponent, {data: this.projects, panelClass: 'custom-dialog-container'});
 
-    dialogRef.afterClosed().subscribe(result => {
+  async getParticipations() {
+    await this.participationService.getParticipations(this.currentUser.id).then(response => {
+      this.participations = response;
+      this.participations.sort((a, b) => a.project.name.toLowerCase() > b.project.name.toLowerCase() ? 1 : -1);
     })
   }
+
+  checkUserProfileValid(): boolean {
+    return this.currentUser.firstName != null &&
+      this.currentUser.lastName != null &&
+      this.currentUser.phoneNumber != null &&
+      this.currentUser.city != null &&
+      this.currentUser.postalCode != null &&
+      this.currentUser.streetAddress != null &&
+      this.currentUser.dateOfBirth != null;
+  }
+
+  async addParticipation() {
+    if (this.checkUserProfileValid()) {
+      await this.projectService.getActiveProjects().then(response => {
+        let projects: Project[] = response;
+        projects.forEach(pro => {
+          this.participations.forEach(par => {
+            if (pro.id == par.project.id) {
+              projects = projects.filter(obj => obj !== pro);
+            }
+          })
+        })
+        let dialogRef = this.dialog.open(AddProjectComponent, {
+          data: projects,
+          width: '350px',
+        });
+        dialogRef.disableClose = true;
+        dialogRef.afterClosed().subscribe(result => {
+          if (result !== 'false') {
+            this.selectedProjects = result;
+            this.selectedProjects.forEach(project => {
+              let participation: Participation = new Participation("00000000-0000-0000-0000-000000000000", this.currentUser, project)
```

Comment:
Ik heb al een aantal keer een empty guid voorbij zien komen. Is dit wat voor een helper class die een empty guid returned?

Reply:
Top suggestie. Zit erin